components/match2/wikis/arenaofvalor/match_legacy.lua

```diff
@@ -0,0 +1,186 @@
+---
+-- @Liquipedia
+-- wiki=arenaofvalor
+-- page=Module:Match/Legacy
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local MatchLegacy = {}
+
+local Json = require('Module:Json')
+local Logic = require('Module:Logic')
+local String = require('Module:StringUtils')
+local Table = require('Module:Table')
+local Variables = require('Module:Variables')
+
+local _GAME_EXTRADATA_CONVERTER = {
+	ban = 'b',
+	champion = 'h',
+}
+
+function MatchLegacy.storeMatch(match2)
+	local match = MatchLegacy._convertParameters(match2)
+
+	match.games = MatchLegacy.storeGames(match, match2)
+
+	MatchLegacy.storeMatchSMW(match, match2)
+
+	return mw.ext.LiquipediaDB.lpdb_match(
+		'legacymatch_' .. match2.match2id,
+		match
+	)
+end
+
+function MatchLegacy._convertParameters(match2)
+	local match = Table.deepCopy(match2)
+	for key, _ in pairs(match) do
+		if String.startsWith(key, 'match2') then
+			match[key] = nil
+		end
+	end
+	match.links = nil
+
+	if String.isNotEmpty(match.walkover) then
+		match.resulttype = match.walkover
+		match.walkover = match.winner
+	end
+
+	match.staticid = 'Legacy_' .. match2.match2id
+
+	-- Handle extradata fields
+	match.extradata = {}
+	local extradata = Json.parseIfString(match2.extradata)
+	match.extradata.matchsection = extradata.matchsection
+	match.extradata.mvpteam = extradata.mvpteam
+	match.extradata.mvp = extradata.mvp
+	match.extradata.comment = extradata.comment
+
+	local opponents = match2.match2opponents or {}
+	match.extradata.team1icon = (opponents[1] or {}).icon
+	match.extradata.team2icon = (opponents[2] or {}).icon
+	match.extradata.opponent1literal = tostring((opponents[1] or {}).type == 'literal')
+	match.extradata.opponent2literal = tostring((opponents[2] or {}).type == 'literal')
```

Comment:
Is it intended to write "true" in this field? Or should the field be populated with the literal, if it's an literal?

Reply:
Should I remove this altogether if its werent used at all? (due to literal was not a supported opponent type) 
![image](https://user-images.githubusercontent.com/88981446/149083049-a8dff681-91a2-49a0-ad00-9b7526e77ba5.png)